builtin/providers/aws/resource_aws_redshift_cluster_test.go

```diff
@@ -104,6 +104,39 @@ func TestAccAWSRedshiftCluster_updateNodeCount(t *testing.T) {
 	})
 }
 
+func TestAccAWSRedshiftCluster_tags(t *testing.T) {
+	var v redshift.Cluster
+
+	ri := rand.New(rand.NewSource(time.Now().UnixNano())).Int()
+	preConfig := fmt.Sprintf(testAccAWSRedshiftClusterConfig_tags, ri)
+	postConfig := fmt.Sprintf(testAccAWSRedshiftClusterConfig_updatedTags, ri)
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAWSRedshiftClusterDestroy,
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: preConfig,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAWSRedshiftClusterExists("aws_redshift_cluster.default", &v),
+					resource.TestCheckResourceAttr(
+						"aws_redshift_cluster.default", "tags.#", "3"),
```

Comment:
Maybe a nitpick, but it would be good to also test the actual tags - e.g. `resource.TestCheckResourceAttr("aws_redshift_cluster.default", "tags.environment", "Production")`